File: Trikeshed/src/jvmMain/java/borg/trikeshed/io/PlatformFile.java
package borg.trikeshed.io;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

public class PlatformFile {
    private final File file;

    public PlatformFile(String path) {
        this.file = new File(path);
    }

    public boolean exists() {
        return file.exists();
    }

    public boolean isDirectory() {
        return file.isDirectory();
    }

    public byte[] readAllBytes() throws IOException {
        return Files.readAllBytes(Paths.get(file.getPath()));
    }
}

File: trikeshed-io/src/jvmMain/kotlin/borg/trikeshed/io/PlatformFile.kt
@file:OptIn(kotlin.ExperimentalUnsignedTypes::class)
package borg.trikeshed.io

import java.io.File

/**
 * JVM implementation of PlatformFile
 */
actual class PlatformFile actual constructor(internal val path: String) {
    internal val file = File(path)
    
    actual fun exists(): Boolean {
        return file.exists()
    }
    
    actual fun isDirectory(): Boolean {
        return file.isDirectory
    }
    
    actual fun readAllBytes(): ByteArray {
        return file.readBytes()
    }
}